### PR TITLE
Fixes #9743: When applying same directive in different rule, reporting is broken

### PR DESCRIPTION
**Pull request by fanf**
https://www.rudder-project.org/redmine/issues/97434

So. I can't reproduce it even with the data. 
The test show that the computed compliance is what is awaited. Not sure what to do with that further more. Merge the PR, mark the problem as "can't reproduce" ?

### CODE REVIEW

File: rudder-core/src/test/scala/com/normation/rudder/services/reports/ComplianceTest.scala [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+        (reports, config)
+      }).openOrThrowException("the test should not throw an exception")
+
+      val runTime = reports.head.executionTimestamp
+      val runinfo = ComputeCompliance(runTime, config, runTime)
+      val status = ExecutionBatch.getNodeStatusReports(config.nodeId, runinfo, reports)
+      println(status.forNode)
+      println(status.runInfo)
+      println(status.statusInfo)
+      println(status.compliance)
```
Comment (ncharles) [inline rudder-core/src/test/scala/com/normation/rudder/services/reports/ComplianceTest.scala]: i think you want to remove these println

File: rudder-core/src/test/scala/com/normation/rudder/services/reports/ComplianceTest.scala [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+ * the corresponding reports, compute the resulting node status, and compare it
+ * to an expected one.
+ *
+ */
+
+
+@RunWith(classOf[JUnitRunner])
+class ComplianceTest extends Specification {
+
+  val dateParser = ((new DateTimeFormatterBuilder())
```
Comment (ncharles) [inline rudder-core/src/test/scala/com/normation/rudder/services/reports/ComplianceTest.scala]: these utility methods may be useful in other tests. Would it make sense to put them in a separate file to be used generically ?

**Comment by fanf** [comment]
PR rebased

File: rudder-core/src/test/scala/com/normation/rudder/services/reports/ComplianceTest.scala [hunk 3]
```diff
@@ -0,0 +1,225 @@
+/*
+*************************************************************************************
+* Copyright 2011 Normation SAS
```
Comment (ncharles) [inline rudder-core/src/test/scala/com/normation/rudder/services/reports/ComplianceTest.scala]: 2016

**Comment by fanf** [comment]
PR rebased

**Comment by fanf** [comment]
OK, merging this PR

**Comment by fanf** [comment]
OK, merging this PR

**Comment by fanf** [comment]
OK, merging this PR

**Comment by fanf** [comment]
OK, merging this PR

**Pull request merged**